internal/config/config.go

```diff
@@ -802,6 +803,46 @@ func (c AzureSEVSNP) EqualTo(old AttestationCfg) (bool, error) {
 	return firmwareSignerCfgEqual && measurementsEqual && bootloaderEqual && teeEqual && snpEqual && microcodeEqual && rootKeyEqual, nil
 }
 
+func (a *AzureSEVSNP) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	aux := &fusedAzureSEVSNP{
+		auxAzureSEVSNP: (*auxAzureSEVSNP)(a),
+	}
+	if err := unmarshal(aux); err != nil {
+		return fmt.Errorf("failed to unmarshal AzureSEVSNP: %w", err)
+	}
+	a = (*AzureSEVSNP)(aux.auxAzureSEVSNP)
+
+	for _, versionType := range []version.Type{version.Bootloader, version.TEE, version.SNP, version.Microcode} {
+		if !convertLatestToNumber(a, versionType, aux) {
+			if !convertStringToUint(a, versionType, aux) {
+				return fmt.Errorf("failed to convert %s version to number", versionType)
+			}
+		}
+	}
```

Comment:
If `convertLatestToNumber` would instead be a `convertLatestToString`, you could call `convertStringToUint` unconditionally, which improve readability a bit.
Not required though, keep what you prefer.

Reply:
I agree but it would imply 2 conversions since changing GetVersion to return a string doesn't make sense. So i rather keep it as is